### DISCUSSION TITLE
How to run Talos without blocked linux capabilities?

### DISCUSSION THREAD

**Discussion by hellt** [siderolabs/talos#8929] [Q&A]
Hi, a weird request maybe, but I would like to use Talos to spin up dev/test k8s clusters using its declarative configuration.
Given that my dev/test clusters run privileged pods that emulate network operating systems, some of them are equipped with `CAP_SYS_BOOT` and talos [prohibits](https://www.talos.dev/v1.7/learn-more/process-capabilities/) those.

Would it be possible to have an option to whitelist those blocked capabilities and enable talos to support dev/test clusters where security is not as important as the easy of installation?

**Comment by hellt** [comment]
I found this helpful thread - https://github.com/siderolabs/talos/issues/6046

I wonder, why setting this in the machine config 

```yaml
machine:
  install:
    extraKernelArgs:
      - sysctl.kernel.kexec_load_disabled=1
```

that is applied as user_data on the machine boot does not have effect? I can only see this kernel flag applied when I do the `talos upgrade`

**Comment by smira** [comment]
You can use `KubeVirt` to run your pods inside a micro-VM with full capabilities enabled, but in general you might expect some restrictions in the containers when it gets to doing a fully privileged operations (like trying to do `kexec` or `reboot` which doesn't make any sense inside the container).

One thing to note here, that CNIs, even those which require deep kernel networking interactions, like Cilium, work perfectly under Talos Linux.

Talos Linux philosophy is being more secure by default, so workloads might need to be adapted to that, and that security enforcement is not something you can turn off. We feel that hardening out of the box is our choice, and unfortunately that might break some workloads in very rare cases.

**Reply by hellt** [reply]
Understood, thanks @smira 
I appreciate having the way to drop the enforced capabilities via the method listed above. 

And I can live with an extra step of a cluster upgrade when I need to support workloads that had sys_boot capabilities assigned to them. 

Having said that we also removing this capability from our container process to not require Talos kernel upgrade
